File: Patterns/Facade Pattern/CSharp/HomeTheater/Tuner.cs
using System;

namespace HomeTheater
{
    public class Tuner
    {
        private string description;
        private double frequency;

        public Tuner(string description)
        {
            this.description = description;
        }

        public void On() => Console.WriteLine($"{description} on");

        public void Off() => Console.WriteLine($"{description} off");

        public void SetFrequency(double frequency)
        {
            this.frequency = frequency;
            Console.WriteLine($"{description} setting frequency to {frequency}");
        }

        public void SetAm() => Console.WriteLine($"{description} setting AM mode");

        public void SetFm() => Console.WriteLine($"{description} setting FM mode");

        public string Description { get => description; private set => description = value; }
    }
}

File: Patterns/Facade Pattern/Cpp/HomeTheater/Tuner.cpp
#pragma once

#include <iostream>
#include <string>

class Tuner {
private:
    string description;
    double frequency;

public:
    Tuner(string description) : description{description} {}
    
    void on() {
        cout << description << " on\n";
    }

    void off() {
        cout << description << " off\n";
    }

    void setFrequency(double frequency) {
        this->frequency = frequency;
        cout << description << " setting frequency to " << frequency << "\n";
    }

    void setAm() {
        cout << description << " setting AM mode\n";
    }

    void setFm() {
        cout << description << " setting FM mode\n";
    }

    string toString() {
        return description;
    }
};